src/smart.ts

```diff
@@ -238,6 +238,11 @@ export async function authorize(
     fhirServiceUrl = url.searchParams.get("fhirServiceUrl") || fhirServiceUrl;
     launch         = url.searchParams.get("launch")         || launch;
 
+    // For these url param inline takes precedence over url parameter
+    if (!client_id) {
+        client_id = url.searchParams.get("client_id")  
+    }
+    
     if (!clientId) {
         clientId = client_id;
     }
```

Comment:
This wouldn't work because `client_id` is a constant and can't be reassigned here. In addition, the `client_id` option is deprecated and `clientId` should be used instead (which is the reason for lines 246 to 248).

Finally, `clientId` should be a `string` or `undefined` (but not `null`), but this will set it to `null` if the url param is not present. We prefer `undefined` because that will make `clientId` to "disappear" from the state after serialization.

With that said, the following code seems to work for me (246 to 248):
```ts
if (!clientId) {
    clientId = client_id || (url.searchParams.get("client_id") ?? undefined);
}
```

Reply:
Thanks for the explanation, it also works for me, however i guess also in the searchparams it should be clientId than, to ensure consistency across the labeling
Pushed the new version